sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -900,6 +902,18 @@ class Analyzer(
             }
           case _ => u
         }
+
+      case u @ UnresolvedTableOrView(identifier) =>
+        expandRelationName(identifier) match {
+          case SessionCatalogAndIdentifier(catalog, ident) =>
+            CatalogV2Util.loadTable(catalog, ident) match {
+              // We don't support view in v2 catalog yet. Here we treat v1 view as table and use
```

Comment:
IIUC, if this one carries a v1Table which is actually a view, then you need to handle the exception in `DataSourceV2Strategy` but why not just fail it here. If so, instead of introducing the new `UnresolvedTableOrView`, I guess `UnresolvedTable` seems to be enough

Reply:
Most of the logic of `UnresolvedTableOrView` is duplicated with `UnresolvedTable `